src/MiddlewareFactory.php

```diff
@@ -107,8 +107,8 @@ public function pipeline(...$middleware): MiddlewarePipe
     {
         // Allow passing arrays of middleware or individual lists of middleware
         if (
-            is_array($middleware[0])
-            && count($middleware) === 1
+            count($middleware) === 1
+            && is_array($middleware[0])
```

Comment:
Since PHP 8.0, splat input data does allow for non-list arguments anyway, so we should check if `$middleware[0]` is actually there with an `isset` or such.

Let's please also add a small test.